src/test/store/receive-profile.test.js

```diff
@@ -1759,18 +1759,35 @@ describe('actions/receive-profile', function() {
         globalTracks,
         rootRange,
       } = await setupWithLongUrl(getSomeProfiles(), {
-        urlSearch1: 'thread=0',
-        urlSearch2: 'thread=1',
+        urlSearch1: 'thread=0&profileName=name 1',
+        urlSearch2: 'thread=1&profileName=name 2',
       });
 
-      const expectedThreads = [profile1.threads[0], profile2.threads[1]].map(
-        (thread, i) => ({
-          ...thread,
-          pid: `${thread.pid} from profile ${i + 1}`,
-          processName: `Profile ${i + 1}: ${thread.name}`,
-          unregisterTime: getTimeRangeForThread(thread, 1).end,
-        })
-      );
+      const profileState1 = stateFromLocation({
+        pathname: '/public/fakehash1/',
+        search: '?profileName=name 1',
+        hash: '',
+      });
+
+      const profileState2 = stateFromLocation({
+        pathname: '/public/fakehash1/',
+        search: '?profileName=name 2',
+        hash: '',
+      });
+
+      const profiles = [
+        { thread: profile1.threads[0], state: profileState1 },
+        { thread: profile2.threads[1], state: profileState2 },
+      ];
+
+      const expectedThreads = profiles.map((profile, i) => ({
+        ...profile.thread,
+        pid: `${profile.thread.pid} from profile ${i + 1}`,
+        processName: `${profile.state.profileName || `Profile ${i + 1}`}: ${
+          profile.thread.name
+        }`,
+        unregisterTime: getTimeRangeForThread(profile.thread, 1).end,
+      }));
```

Comment:
I think this is more complex than it should. And actually the previous code was probably also more complex than it could!

I think you could simply do something like:
```js
const expectedThreads = [{
  ...profile1.threads[0],
  pid: ...
  processName: "Actual Expected Name",
  ...
},
{
  ...profile2.threads[0],
  pid: ...
  processName: "Actual Expected Name",
  ...
},
// comparison thread
expect.objectContaining({
  processType: 'comparison',
  pid: 'Diff between 1 and 2',
  name: 'Diff between 1 and 2',
})];
```
(the last part replaces the line `expectedThreads.push(` below)

A good test is usually more explicit and less "smart", and I believe it was too smart before.

Reply:
Hi, @julienw thanks for showing me a simpler solution for the test. I have made the changes 😄